Sources/Common/Service/HttpRequestError.swift

```diff
@@ -14,6 +14,8 @@ public enum HttpRequestError: Error {
     case noOrBadNetwork(_ urlError: URLError)
     /// Request was not able to get a response from server. Maybe no network connection?
     case noRequestMade(_ error: Error?)
+    /// 400 status code. The request body contains errors or is malformed.
+    case badRequest
```

Comment:
I disagree with this. See my comment: https://github.com/customerio/customerio-android/pull/177#discussion_r1123134468

Reply:
I am okay with any, will prefer `badRequest` as well, as it looks more natural to me. So if Levi, is okay, we can but if they feel its something that can really improve readability then we can go that way.

Reply:
I just wanted to share my opinion here. As I think it is little too much detail in name and not really needed. Because in my mind, when I see `badRequest` in any HTTP class, it automatically relates to `400` because of being standard practice. 

I would prefer not to include the code in name here, but I'm also okay with continuing what you both folks think looks better. And whatever we decide for this, we should keep the same for Android as well.

Reply:
I would agree that naming the enum with 400 and with `bad request` feels verbose and not adding value. 

My intention with the naming is to make sure that "bad request" is *a server response* and not a generic "bad HTTP request" that the SDK made. When someone says, "That was a bad request" maybe they meant that the SDK has a bug in it and the request "was bad" and a request wasn't even made. Maybe they meant the SDK made a successful request but the server responded back with a 400 response. 

I am OK with a different name that doesn't feel verbose. My concern with the name "bad request" is that it could be interrupted in a wrong way by someone reading the code. As long as the name is showing that a response came back from the server and the response was a 400 response, I am happy with the name 👍.